aiohttp/web_urldispatcher.py

```diff
@@ -1257,7 +1257,21 @@ def _quote_path(value: str) -> str:
 
 
 def _unquote_path(value: str) -> str:
-    return URL.build(path=value, encoded=True).path.replace("%2F", "/")
```

Comment:
Oh, wait, we were creating a URL here, now we're not...
Ideally we'd want to utilise URL.path in some way though, which is already optimised.

Reply:
I'm not sure this is fixable because in this case `value` is already unquoted and when we run the regex we get the `%2F` in the match group but we don't know if its an encoded value or if its yarl skipping decoding `%2F` since its already decoded.

Reply:
As far as I can see, both calls to this function are actually passing the value from `request.rel_url.path`, so it looks to me like it should already be safe?

Reply:
I would expect the second one to return `'p%2Fa%2Ft%2Fh%3D1'` and not `'p/a/t/h%3D1'`

Reply:
I see. Maybe we can add something to yarl...?

Reply:
I think if we revert the change in yarl everything will work as expected with this PR

Reply:
Have you tried it? We made that change to fix the routing, I'm not sure I see any changes here to make that routing work..

Reply:
Yes. I did have to change this test https://github.com/aio-libs/aiohttp/pull/9267/files#diff-ffe45e5e9ca19ae2918aa4ac4796a9871765331a2b862c9c5efc898e36ccc068R773

I looked at all the views in Home Assistant and found they were all made before the original regression so they have `:.+` in them

Reply:
I reverted the commits that require a new yarl and doing a CI run with 1.9.4 to see if it passes still

Reply:
We have `app.router.add_route("GET", "/resource2/a/b%2Fc/", handler)` in the middleware tests

Reply:
https://github.com/aio-libs/aiohttp/pull/3051

Reply:
The middleware check expects paths to be raw
https://github.com/aio-libs/aiohttp/blob/203ca4273edcd6ea5372c630b054381ebb3bacdb/aiohttp/web_middlewares.py#L107